command/command.go

```diff
@@ -259,35 +270,40 @@ func handleCommitRequest(cache *cache.Cache, commitMsg *sync_pb.CommitMessage, c
 		}
 
 		if !isUpdateOp { // Create
-			if currentNormalItemCount+currentHistoryItemCount+newNormalCount+newHistoryCount >= maxClientObjectQuota {
+			if currentNormalItemCount+currentHistoryItemCount+newNormalCount+newHistoryCount >= maxClientObjectQuota+boostedQuotaAddition {
 				rspType := sync_pb.CommitResponse_OVER_QUOTA
 				entryRsp.ResponseType = &rspType
 				entryRsp.ErrorMessage = aws.String(fmt.Sprintf("There are already %v non-deleted objects in store", currentNormalItemCount+currentHistoryItemCount))
 				continue
 			}
 
-			conflict, err := db.InsertSyncEntity(entityToCommit)
-			if err != nil {
-				log.Error().Err(err).Msg("Insert sync entity failed")
-				rspType := sync_pb.CommitResponse_TRANSIENT_ERROR
-				if conflict {
-					rspType = sync_pb.CommitResponse_CONFLICT
+			if !isHistoryRelatedItem || currentHistoryItemCount+newHistoryCount < maxClientHistoryObjectQuota {
+				// Insert all non-history items. For history items, ignore any items above history quoto
+				// and lie to the client about the objects being synced instead of returning OVER_QUOTA
```

Comment:
>and lie to the client about the objects being synced instead of returning OVER_QUOTA

as I recall, the number of entries at commit request must match the number of entries at response
https://source.chromium.org/chromium/chromium/src/+/main:components/sync/protocol/sync.proto;l=339;drc=8b35c4f4d196c367754b2fedd213b662c98f34fd;bpv=0;bpt=0 

I can't understand this  by the code, but it seems the number of entries in response will be less than in request? 
If yes, I need to test in by running with client.

Reply:
We still return successful response entries for the omitted items, hence the "lie". I updated the dev environment with a tiny history quota of 25 items, feel free to try it out